### PR TITLE
feat: refactor CI/CD

### PR DESCRIPTION
**Pull request by smuu**
Tracked by epic: https://github.com/celestiaorg/devops/issues/111

Changes:
- remove unused workflows
- use of common workflow for container build (mockserv)
- let dependabot bump Dockerfile

Please comment on all ideas, thoughts, and questions as a comment.

Closes https://github.com/celestiaorg/devops/issues/163

### CODE REVIEW

File: .github/dependabot.yml
Comment (MSevey) [inline .github/dependabot.yml:24]: I'm not sure the impact of this. Will this overwrite any other review request automation that is set up?

I'm wondering if it is better to remove this and keep all the review request logic in a single source of truth for the repo. 

I also haven't used this feature of dependabot, and I don't think we use it anywhere else currently so it might just add some noise

File: .github/workflows/lint.yml
Comment (MSevey) [inline .github/workflows/lint.yml:17]: [question] have there been instances of this lint hanging? just curious as the rationale for adding the timeout.

File: .github/workflows/lint.yml
Comment (MSevey) [inline .github/workflows/lint.yml:34]: Should the over arching timeout be 10m as well then instead of 8?  Or we can remove one as I wouldn't expect any of the other steps to timeout.

File: .github/workflows/lint.yml
Comment (MSevey) [inline .github/workflows/lint.yml:36]: where is this getting defined? I don't see an `env` section nor a statement about secrets being inherited.

**Review by MSevey — COMMENTED** [review]
Overall looks good, a few question/comments, plus it would be good to fix the golangci lint errors that are showing in the unchanged files.

File: .github/dependabot.yml
Comment (tzdybal) [reply .github/dependabot.yml:24]: I think it only adds docker-related dependabot checks for files in `/docker` directory.

File: .github/workflows/lint.yml
Comment (tzdybal) [reply .github/workflows/lint.yml:17]: meme

File: .github/workflows/lint.yml
Comment (tzdybal) [reply .github/workflows/lint.yml:36]: Is it set by `get-diff-action`?

**Review by tzdybal — COMMENTED** [review]
Left some comments.

File: .github/workflows/lint.yml
Comment (tzdybal) [reply .github/workflows/lint.yml:17]: meme

File: .github/workflows/docker-build-da-mockserv.yml
Comment (smuu) [reply .github/workflows/docker-build-da-mockserv.yml:1]: https://celestia-team.slack.com/archives/C04UH5EKCP5/p1680520061351889?thread_ts=1680510646.603789&cid=C04UH5EKCP5

@Manav-Aggarwal Do you want to use the mockserv or local-celestia-devnet?

File: .github/dependabot.yml
Comment (smuu) [reply .github/dependabot.yml:24]: I copy & pasted it from the current setup which was introduced here: https://github.com/rollkit/rollkit/commit/5bf2de6f8be96e8144ff8ecefe23cde1e85c1dbf

I'm not aware of the request automation in github. If there is some automation that adds the rollkit team as a reviewer I'm okay with removing it from the dependabot file.

File: .github/workflows/lint.yml
Comment (smuu) [reply .github/workflows/lint.yml:36]: https://github.com/technote-space/get-diff-action/tree/v6.1.2/

The `technote-space/get-diff-action` set's the `GIT_DIFF` environment in the current job so it can be used by another step.

File: .github/workflows/lint.yml
Comment (smuu) [reply .github/workflows/lint.yml:17]: Removed this in favor of the timeout in `golangci/golangci-lint-action`. Other steps should not hang and if the timeout of the lint actions hit the job can terminate without a job timeout.

**Review by MSevey — APPROVED** [review]
LGTM, ping me when you get a second approver and I will admin merge and update the branch protection rules for the changes in CI

**Pull request merged**
